templates/vsphere/template-variables.rc

```diff
@@ -0,0 +1,22 @@
+# Source: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/docs/getting_started.md
+
+export VSPHERE_USERNAME="vi-admin@vsphere.local"                    # The username used to access the remote vSphere endpoint
+export VSPHERE_PASSWORD="admin!23"                                  # The password used to access the remote vSphere endpoint
+
+export VSPHERE_SERVER="10.0.0.1"                                    # The vCenter server IP or FQDN
+export VSPHERE_DATACENTER="SDDC-Datacenter"                         # The vSphere datacenter to deploy the management cluster on
+export VSPHERE_DATASTORE="DefaultDatastore"                         # The vSphere datastore to deploy the management cluster on
+export VSPHERE_NETWORK="VM Network"                                 # The VM network to deploy the management cluster on
+export VSPHERE_RESOURCE_POOL="*/Resources"                          # The vSphere resource pool for your VMs
+export VSPHERE_FOLDER="vm"                                          # The VM folder for your VMs. Set to "" to use the root vSphere folder
+export VSPHERE_TEMPLATE="ubuntu-1804-kube-v1.17.3"                  # The VM template to use for your management cluster.
+export CONTROL_PLANE_ENDPOINT_IP="192.168.9.230"                    # the IP that kube-vip is going to use as a control plane endpoint
+export VIP_NETWORK_INTERFACE="ens192"                               # The interface that kube-vip should apply the IP to. Omit to tell kube-vip to autodetect the interface.
+export VSPHERE_TLS_THUMBPRINT="..."                                 # sha256 thumbprint of the vcenter certificate: openssl x509 -sha256 -fingerprint -in ca.crt -noout
+export EXP_CLUSTER_RESOURCE_SET="true"                              # This enables the ClusterResourceSet feature that we are using to deploy CSI
+export VSPHERE_SSH_AUTHORIZED_KEY="ssh-rsa AAAAB3N..."              # The public ssh authorized key on all machines in this cluster.
+                                                                    #  Set to "" if you don't want to enable SSH, or are using another solution.
+export VSPHERE_STORAGE_POLICY=""                                    # This is the vSphere storage policy. Set it to "" if you don't want to use a storage policy.
+export "CPI_IMAGE_K8S_VERSION"="v1.28.0"                            # The version of the vSphere CPI image to be used by the CPI workloads
```

Comment:
let's set the whole image as an argument here, same for csi

Reply:
Smart